packages/client/wallets/smart-wallet/src/blockchain/wallets/EVMSmartWallet.ts

```diff
@@ -126,6 +126,16 @@ export class EVMSmartWallet {
         return this.crossmintService.fetchNFTs(this.address, this.chain);
     }
 
+    /**
+     * Sends a contract call transaction and returns the hash of a confirmed transaction.
+     * @param address the address of the contract to be called
+     * @param abi the ABI of the contract - ***should be defined as a typed variable*** to enable type checking of the contract arguments, see https://viem.sh/docs/typescript#type-inference for guidance
+     * @param functionName the name of the function to be called
+     * @param args the arguments to be passed to the function
+     * @param resendCallback a callback function that will be called if the transaction needs to be resent. Returns a value that controls transaction rebroadcasting See the type of `TransactionServiceResendCallback` for more information.
+     * @returns The transaction hash.
+     * @throws `SendTransactionError` if the transaction fails to send. Will be subclass `SendTransactionExecutionRevertedError` if the transaction reverted either during simulation or on-chain.
```

Comment:
Can we expand this into two @throws lines?

SendTransactionExecutionRevertedError: if the transaction reverted during simulation or on chain.
SendTransactionError: ???

also dont we have network errors, and other reasons why a TX may not be sent?

Reply:
> also dont we have network errors, and other reasons why a TX may not be sent?
This is the benefit of wrapping the errors, we can choose which errors (like execution reverted) we want to have special handling for (e.g. adding tenderly links in the future) while also not having to write a custom error for everything that can fail when sending a transaction, which is an extremely difficult task